corehq/apps/api/resources/v0_1.py

```diff
@@ -270,7 +270,7 @@ class Meta(UserResource.Meta):
 
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
-        username = bundle.request.GET.get('username')
+        username = bundle.request.GET.get('web_username')
```

Comment:
@czue Yeah it doesn't affect basic or digest. I checked on proxy logs, there are no real requests to `web-user` endpoint apart from (me and ethan for testing). The log file is since 13th May. Should I be checking even older logs?

@millerdev The only reason I am doing in this PR is because the bug is uncovered due to tests written in this PR. The bug is related to this PR